diff --git a/src/tls/tls_misc.c b/src/tls/tls_misc.c
index b7acd1d..83f4d85 100644
--- a/src/tls/tls_misc.c
+++ b/src/tls/tls_misc.c
@@ -1402,10 +1402,7 @@ void    tls_check_version(void)
 	    && (lib_info.micro < hdr_info.micro
 		|| hdr_info.major == 0
 		|| (hdr_info.major == 1 && hdr_info.minor == 0))))
-	msg_warn("run-time library vs. compile-time header version mismatch: "
-	     "OpenSSL %d.%d.%d may not be compatible with OpenSSL %d.%d.%d",
-		 lib_info.major, lib_info.minor, lib_info.micro,
-		 hdr_info.major, hdr_info.minor, hdr_info.micro);
+	; // Nothing. Warning output removed.
 }
 
 /* tls_compile_version - compile-time OpenSSL version */
